docs/rfcs/2023-10-12-remote-ui.md

```diff
@@ -0,0 +1,88 @@
+# Remote desktop / mobile implementation details
+
+This follows the previous RFC for [remote profiles](./2023-09-12-remote-profile.md).
+
+Code uses terms remote controller and remote host to designate the roles, and CLI client can support both.
+
+This document uses the terms "mobile" to mean remote host and "desktop" to mean remote controller, mobile apps will only support "remote host" role (via UI, and, possibly, via the compilation flag to remove this functionality from the code), and desktop will only support "remote controller" role.
+
+## Mobile (host)
+
+UX is described [here](./2023-09-12-remote-profile.md#on-a-mobile-device).
+
+When mobile is connected to the remote desktop it will receive no events from the core (other than remote session connection status changes), and the core will accept no commands from the UI (other than to terminate the remote session).
+
+As currently mobile maintains the list of connection statuses for all profiles, this state will have to be maintained in Haskell to be able to send it to the remote desktop UI when it is connected. It will also be used to update mobile UI when control is returned to the mobile.
+
+The switching between remote host role and local UI role should prevent the possibility of any race conditions.
+
+To swith to remote host role:
+- UI: will show the screen with "connected" status and "cancel" button, with disabled sleep timer on iOS, as iOS app will have to stay in foregro. Android will be able to function in background in this state.
+- core: stop adding events to the output queues (via some flag).
+- UI: process remaining events in the output queue and stop receiving them.
+- core: stop sending events to and accepting commands from UI.
+- core: send current list of profiles, coversations, and connections statuses to the remote desktop.
+- core: start sending events to and accepting commands from remote desktop.
+- core: start adding events to remote output queue.
```

Comment:
how "adding events to remote output queue" is different from "sending event to / remote desktop"?..